### PR TITLE
Update file header comments and AUTHORS file

### PR DESCRIPTION
**Pull request by beru**
Hi, this PR updates file header comments and AUTHORS file.
#228 is related.

### CODE REVIEW

**Comment by Randl** [comment]
Can you add `#pragma once` here:
https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/io/display.h#L3

**Comment by beru** [comment]
@Randl 
> Can you add #pragma once here:
> https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/io/display.h#L3

Sure.

**Comment by Randl** [comment]
I believe legally it should be `Copyright (c) 2017, Taiga Nomi and the respective contributors`, since the rights weren't passed.

**Comment by beru** [comment]
> I believe legally it should be Copyright (c) 2017, Taiga Nomi and the respective contributors, like https://github.com/tiny-dnn/tiny-dnn/pull/677/files#diff-39eecafba31d7e12041d2fcec3915017L9 , since the rights weren't passed.

Updated.

**Comment by edgarriba** [comment]
just some LICENCE examples

https://github.com/dmlc/mxnet/blob/master/LICENSE
https://github.com/opencv/opencv/blob/master/LICENSE
https://github.com/tensorflow/tensorflow/blob/master/LICENSE
https://github.com/pytorch/pytorch/blob/master/LICENSE

I'd add something like that:

```
Copyright (c) 2013-2017 Taiga Nomi 
Copyright (c) 2015-2017 The respective contributors
```

check https://github.com/tiny-dnn/tiny-dnn/graphs/contributors
until 2015 just @nyanp contributed to the project

**Comment by beru** [comment]
@edgarriba I wonder if it really needs to be very specific.
And do we have to rewrite it each coming year?
Is it the final decision?
Maybe it's time for me to learn automating update file headers. :tired_face:

**Comment by beru** [comment]
@nyanp Can I uniformly use `Copyright (c) 2013, Taiga Nomi and the respective contributors`?
Checking each file's first publication year is just hard...

**Comment by beru** [comment]
@nyanp Is a comma after the year needed?
> If we apply this rule, we need to write copyright like
> `Copyright (c) xxxx, Taiga Nomi and the respective contributors`

**Comment by beru** [comment]
@edgarriba Thank you for introducing the LICENSE examples.

`pytorch` doesn't use file header copyright comment.

`mxnet` uses short file header comment, `Copyright (c) 2015 by Contributors` line and `\file` and `\header` lines follow.

A quote from this page https://hackerboss.com/get-rid-of-templates/
> If you really feel you must at least refer to a license in open source code, you should be fine with just one line in each file.
> If you do this, I don’t much care what you then put in the “LICENSE” file.  You can put in all the copyright assertions and years for all I care.  Just stop being a monkey; don’t borify the source code with attention stealing enterprisey legalese and copyrights.
> `/* See the file "LICENSE" for the full license governing this code. */`

**Comment by Randl** [comment]
ping @edgarriba @nyanp @beru @bhack 

Can we stick with `Copyright (c) 2013, Taiga Nomi and the respective contributors` for all current and future files?

```Each contributor holds copyright over their respective contributions.
The project versioning (Git) records all such contribution source information.```
might be added too.

**Comment by edgarriba** [comment]
@Randl that should work, however @beru then you should replace `2017 -> 2013`

**Comment by beru** [comment]
Then I'm going to use this one.
```
Copyright (c) 2013, Taiga Nomi and the respective contributors
```
I'll update the code tomorrow.

**Comment by nyanp** [comment]
> Then I'm going to use this one.
> Copyright (c) 2013, Taiga Nomi and the respective contributors

Ok, go ahead 👍

**Comment by Randl** [comment]
looks like someone rebased and didn't force pushed

**Comment by beru** [comment]
@Randl It took some time to invent a time machine and update the comments.

**Comment by Randl** [comment]
I afraid you somehow ruined formatting too

**Comment by beru** [comment]
@Randl Thank you for the notification. I wish the latest commit would be alright.

File: tiny_dnn/core/framework/device.fwd.h [hunk 1]
```diff
@@ -1,51 +1,9 @@
...
-    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
-    FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-    DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-    SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-   LIABILITY,
-    OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
-   USE
-    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    Use of this source code is governed by a BSD-style license that can be found
```
Comment (edgarriba) [inline tiny_dnn/core/framework/device.fwd.h]: @beru remove this couple of lines

**Pull request merged**